### PR TITLE
Add Logistic Regression

### PR DESCRIPTION
**Pull request by ynXiang**
This PR adds Logistic Regression (lr) to sparkbench of Hibench. It contains a data generator to generate RDD[LabeledPoint] data based on random balance decision tree, and a LR algorithm with LBFGS using these data.

### CODE REVIEW

**Comment by ynXiang** [comment]
I also add ALS workload

File: conf/workloads/ml/kmeans.conf [hunk 1]
```diff
@@ -53,4 +53,4 @@ hibench.kmeans.input.sample		${hibench.workload.input}/samples
 hibench.kmeans.input.cluster		${hibench.workload.input}/cluster
 
 hibench.workload.input			${hibench.hdfs.data.dir}/Kmeans/Input
-hibench.workload.output			${hibench.hdfs.data.dir}/Kmeans/Output
```
Comment (VinceShieh) [inline conf/workloads/ml/kmeans.conf]: it's weird to see this change

**Comment by VinceShieh** [comment]
@ynXiang pls provide a description for this PR.

**Comment by VinceShieh** [comment]
LGTM

File: conf/workloads/ml/lr.conf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+hibench.lr.huge.maxresultsize                   "3g"
+hibench.lr.gigantic.examples                    10000
+hibench.lr.gigantic.features                    500000
+hibench.lr.gigantic.frac_categorical_features   0
+hibench.lr.gigantic.frac_binary_features        0
+hibench.lr.gigantic.maxresultsize               "5g"
+hibench.lr.bigdata.examples                    	10000
+hibench.lr.bigdata.features                    	700000
+hibench.lr.bigdata.frac_categorical_features   	0
+hibench.lr.bigdata.frac_binary_features        	0
```
Comment (carsonwang) [inline conf/workloads/ml/lr.conf]: `frac_categorical_features` and `frac_binary_features` are always 0 for all scales?

Comment (VinceShieh) [inline conf/workloads/ml/lr.conf]: yes, these flags control the feature type (binary/categorical/continuous), for LR we now set all the data with continuous type. Suggest keep those flags, as maybe sometime in future we will test on other types of data, then we can simply change the configuration flile without having to touch the source code.

**Comment by carsonwang** [comment]
Merged. Thanks.

**Pull request merged**